lib/ui/recommendation.js

```diff
@@ -141,15 +141,25 @@ Recommendation.prototype = {
       io: this.io,
       win: this.win
     }).render();
+
     if (this.el) {
-      if (this.el.firstChild) {
-        this.el.replaceChild(recommendation, this.el.firstChild);
+      if (recommendation) {
+        if (this.el.firstChild) {
+          if (urlOf(this.el.firstChild) !== urlOf(recommendation)) {
+            this.el.replaceChild(recommendation, this.el.firstChild);
```

Comment:
We're already doing a similar check on the JS side: https://github.com/mozilla/universal-search/blob/master/lib/ui/recommendation.js#L130-L136

How about updating the existing duplicate check to just compare URLs?